### PR TITLE
Align the length passed to __syscall_mmap2() so the struct map* struc…

### PR DESCRIPTION
**Pull request by vargaz**
…ture is aligned properly.

### CODE REVIEW

**Review by kripken — APPROVED** [review]
Thanks!

File: system/lib/libc/emscripten_mmap.c [hunk 1]
```diff
@@ -126,13 +128,14 @@ intptr_t __syscall_mmap2(intptr_t addr, size_t len, int prot, int flags, int fd,
   // but it is widely used way to allocate memory pages on Linux, BSD and Mac.
   // In this case fd argument is ignored.
   if (flags & MAP_ANONYMOUS) {
+    size_t alloc_len = ALIGN_TO(len, 16);
```
Comment (sbc100) [inline system/lib/libc/emscripten_mmap.c:131]: Can this be `alignof(struct map)` instead or a magic constant?

Comment (vargaz) [reply system/lib/libc/emscripten_mmap.c:131]: It only needs to be aligned to 4/8 tbh, so the fields of the struct are aligned.

Comment (vargaz) [reply system/lib/libc/emscripten_mmap.c:131]: Yes.

**Pull request merged**